Assets/SEE/Controls/Actions/ResizeNodeAction.cs

```diff
@@ -0,0 +1,798 @@
+            /// </summary>
+            /// <param name="direction">The direction for which the handle is created.</param>
+            /// <param name="parentWorldPosition">The cached parent world-space position.</param>
+            /// <param name="parentWorldScale">The cached parent world-space scale.</param>
+            private GameObject CreateHandle(Vector3 direction, Vector3 parentWorldPosition, Vector3 parentWorldScale)
+            {
+                GameObject handle = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                handle.GetComponent<Renderer>().material.color = handleColor;
+                handle.transform.localScale = handleScale;
+                handle.transform.localPosition = parentWorldPosition + 0.5f * Vector3.Scale(parentWorldScale, direction);
+                handle.name = $"handle__{direction.x}_{direction.y}_{direction.z}";
+                handle.transform.SetParent(transform);
+                return handle;
+            }
+
+            /// <summary>
+            /// Starts a resize step if a handle is selected.
+            /// <para>
+            /// If the user points to a handle, the <see cref="currentResizeStep"/> attribute is set
+            /// to the respective direction vector and the current transform values.
+            /// </para>
+            /// </summary>
+            void StartResizing()
+            {
+                if (!Raycasting.RaycastAnything(out RaycastHit hit))
+                {
+                    return;
+                }
+                Vector3? resizeDirection = handles.TryGetValue(hit.collider.gameObject, out Vector3 value) ? value : null;
+                if (resizeDirection == null)
+                {
+                    return;
+                }
+
+                currentResizeStep = new (hit.point, resizeDirection.Value, transform);
+            }
+
+            /// <summary>
+            /// Does the actual resizing (scaling and positioning).
+            /// </summary>
+            void UpdateSize()
+            {
+                Raycasting.RaycastLowestNode(out RaycastHit? targetObjectHit, out Node hitNode, nodeRef);
+                if (!targetObjectHit.HasValue)
+                {
+                    return;
+                }
+
+                // Collect siblings
+                Transform parent = transform.parent;
+                // We use an initial size of the list so that the memory does not need to get
+                // reallocated each time an item is added.
+                List<Transform> siblings = new (parent.childCount);
+                foreach (Transform sibling in parent)
+                {
+                    if (sibling != transform && sibling.gameObject.IsNodeAndActiveSelf())
+                    {
+                        siblings.Add(sibling);
+                    }
+                }
+                // Debug.Log("Siblings: " + string.Join(", ", siblings.Select(item => item.name)));
```

Comment:
I kept this around for easier debugging, so that one does not have to struggle with the string conversion later.
If it feels too clumsy, please feel free to remove it.